syntax = "proto3";

// The greeting service definition.
service Imager {
 rpc UploadFile(stream UploadFileRequest) returns (StringResponse) {}
}

message StringResponse {
  string message = 1;
}

message MetaData {
  string filename = 1;
  string extension = 2;
}

message UploadFileRequest {
  oneof request {
    MetaData metadata = 1;
    bytes chunk_data = 2;
  }
}